File: targets/java/javalin/src/main/java/rb/javalin/routes/Authorized.java
package rb.javalin.routes;

import io.javalin.config.JavalinConfig;
import rb.domain.Domain;

/**
 * authorized: the framework's authorization mechanism, crypto excluded.
 *
 * A before handler bound to this path, not an if in the handler. An if would measure the
 * language; the point of the family is the framework's own plumbing. Javalin refuses a
 * request from a before handler by skipping ahead, which is what the exception does.
 */
public final class Authorized {
  private Authorized() {}

  public static void register(JavalinConfig cfg) {
    cfg.routes.before("/authorized/small", ctx -> {
      if (!Domain.tokenOk(ctx.header("authorization"))) {
        ctx.status(403).json(Domain.forbiddenBody()).skipRemainingHandlers();
      }
    });

    // rb:snippet authorized.allowed authorized.denied
    cfg.routes.get("/authorized/small", ctx -> ctx.json(Domain.payload("small")));
  }
}

File: targets/dotnet/wolverine-http/Routes/Authorized.cs
using RequestBench.Domain;
using Wolverine.Attributes;
using Wolverine.Http;

namespace RequestBench.WolverineTarget.Routes;

/// <summary>
/// authorized: the framework's authorization mechanism, crypto excluded.
///
/// Middleware named on the one endpoint, not an if in the handler. An if would measure the
/// language; the point of the family is the framework's own plumbing. A Before that returns
/// a result stops the chain, which is how Wolverine refuses a request.
/// </summary>
// rb:wiring authorized.*
public static class RequireToken
{
    public static IResult? Before(HttpRequest request, DomainModel domain) =>
        domain.TokenOk(request.Headers.Authorization) ? null : Results.Problem(statusCode: 403);
}

public static class AuthorizedEndpoints
{
    [WolverineGet("/authorized/small")]
    [Middleware(typeof(RequireToken))]
    public static PayloadBody Small(DomainModel domain) => domain.Payload("small");
}
